src/tpke_pubkey.erl

```diff
@@ -1,26 +1,32 @@
 -module(tpke_pubkey).
 
 -record(pubkey, {
-          players,
-          k,
-          g1,
-          g2,
-          verification_key,
-          verification_keys
+          players :: pos_integer(),
+          k :: non_neg_integer(),
+          g1 :: erlang_pbc:element(),
+          g2 :: erlang_pbc:element(),
+          verification_key :: erlang_pbc:element(),
+          verification_keys :: [erlang_pbc:element(), ...]
          }).
 
 -type pubkey() :: #pubkey{}.
+%% XXX: this is the {U, V, W} tuple, probably name it better?
+-type encrypted() :: {erlang_pbc:element(), binary(), erlang_pbc:element()}.
 
--export_type([pubkey/0]).
+-export_type([pubkey/0, encrypted/0]).
 -export([init/6, lagrange/3, encrypt/2, verify_ciphertext/2, verify_share/3, combine_shares/3, hash_message/2, verify_signature/3, combine_signature_shares/2, verify_signature_share/3, deserialize_element/2]).
 
 -export([hashH/2]).
 
+%% XXX: I suppose K can be 0 here?
+-spec init(pos_integer(), non_neg_integer(), erlang_pbc:element(), erlang_pbc:element(), erlang_pbc:element(), [erlang_pbc:element(), ...]) -> pubkey().
 init(Players, K, G1, G2, VK, VKs) ->
     #pubkey{players=Players, k=K, verification_key=VK, verification_keys=VKs, g1=G1, g2=G2}.
 
 %% Section 3.2.2 Baek and Zheng
 %% Epk(m):
+%% XXX: why is V a binary? by design or am I mistaken
```

Comment:
V is a binary by design, according to the paper... it's some data XOR'd with the original message.